lib/openldap.c

```diff
@@ -456,163 +599,147 @@ static CURLcode oldap_done(struct Curl_easy *data, CURLcode res,
   return CURLE_OK;
 }
 
+static CURLcode client_write(struct Curl_easy *data, ...)
+{
+  char *s;
+  va_list ap;
+  CURLcode result = CURLE_OK;
+
+  va_start(ap, data);
+  for(s = va_arg(ap, char *); s; s = va_arg(ap, char *)) {
+    size_t len = va_arg(ap, size_t);
+
+    if(len) {
+      result = Curl_client_write(data, CLIENTWRITE_BODY, s, len);
+      if(result)
+        break;
+      data->req.bytecount += len;
+    }
+  }
+  va_end(ap);
```

Comment:
This function looks very error prone... I really think it needs to be made simpler. Perhaps just like:
~~~c
static CURLcode client_write(struct Curl_easy *data,
                             char *prefix,
                             char *value, size_t len,
                             char *suffix)
{
}

Reply:
Yes, doable! Considering its current use, a lesser universal implementation is sufficient.

Reply:
Done, but the char pointers have to be const as literals are passed.
BTW: is there a known reason why `Curl_client_write` does not declare its `ptr` argument as const?

Reply:
Yes, look at the code. It calls functions that potentially replaces contents of the buffer **in place**:

https://github.com/curl/curl/blob/44f88e662fea270de13927333dd7e2030363447e/lib/sendf.c#L628-L641

Reply:
Ah, so this is for FTP only. Thanks for pointing this out.